Tests/MongoSwiftTests/DocumentTests.swift

```diff
@@ -389,17 +403,45 @@ final class DocumentTests: XCTestCase {
         doc["int64"] = Int64.min
         expect(doc.data).to(equal(pointer))
 
+        let newOid = ObjectId()
+        doc["oid"] = newOid
+        expect(doc.data).to(equal(pointer))
+
+        doc["timestamp"] = Timestamp(timestamp: 5, inc: 10)
+        expect(doc.data).to(equal(pointer))
+
+        doc["datetime"] = Date(msSinceEpoch: 2000)
+        expect(doc.data).to(equal(pointer))
+
         // return early as we will to use an Int requiring > 32 bits after this 
         if XCTestCase.is32Bit {
-            expect(doc).to(equal(["double": 3.0, "int32": 20, "int64": Int64.min, "bool": true, "decimal": Decimal128("100")]))
+            expect(doc).to(equal([
```

Comment:
This comment is intended for comment two lines up. This doesn't actually return early. Should the comment be updated?

Reply:
Good catch. However, I think the right thing to do is keep the comment but correct the code to match the comment. The right thing to do here definitely seems to be to return early if we are on a 32 bit system. I will try to see if I can use the Github suggestion thing to chuck in a `return`, and if you think that sounds good, you can approve it?

Reply:
From seeing the logic, it seems like it should work fine on a 32-bit system too. I've made it so that we always run the check and then return if the system is 32 bit.